cmd/subctl/join.go

```diff
@@ -69,6 +71,10 @@ var joinCmd = &cobra.Command{
 		clientProducer, err := client.NewProducerFromRestConfig(clientConfig)
 		exit.OnError(status.Error(err, "Error creating the client producer"))
 
+		networkDetails := getNetworkDetails(clientProducer, status)
```

Comment:
I find 2 issues with this:
1. I strongly believe that `cmd/subctl` should just strictly be `cobra` stuff and nothing else.
2. we are expecting `pkg/join` to know `*CIRD` before hand. So `join` is not smart enough to figure out these details. Any third part code using `join`  would also have to pre-determine these values and can't reply on `join`.

Reply:
Where should `survey`-based code reside? The library itself can’t rely on that, since it can’t expect to have any way to interact with an end-user.

Reply:
> I find 2 issues with this:
> 
> 1. I strongly believe that `cmd/subctl` should just strictly be `cobra` stuff and nothing else.

By that I assume you mean processing arguments and prompting the user.  That's what this code is doing. We call `network.Discover` here in order to determine if we should prompt the user for any CIDRs if one is not provided.

> 2. we are expecting `pkg/join` to know `*CIRD` before hand. So `join` is not smart enough to figure out these details. Any third part code using `join`  would also have to pre-determine these values and can't reply on `join`.

They can by simply calling `network.Discover`. But they don't need to do this. If the CIDRs aren't provided then the operator controller will determine them via `network.Discover`.

Reply:
`survey-` code should reside here. It should ask for CIDR here. `join` should `network.Discover`, get the CIDR and warn users if it is not same as user-specified adn update the `ServiceCIDRAutoDetected` accordingly. But I see you removed that field completely.

Reply:
> `survey-` code should reside here. It should ask for CIDR here. `join` should `network.Discover`, get the CIDR and warn users if it is not same as user-specified 

That's what we're doing here. We also do the same check in the operator. It would be wasteful and unnecessary to perform `network.Discover` twice during `subctl join`. We don't currently have any other users of this API so I think it's reasonable at this point to assume the caller has done their due diligence wrt the CIDRs. We can always revisit based on feedback if we ever do have another user. Also the operator does this same check and logs a warning anyway.

>adn update the `ServiceCIDRAutoDetected` accordingly. But I see you removed that field completely. So I guess, it won't make sense to just warn the users if the provided and detected values won't match.

`ServiceCIDRAutoDetected` was used to pass an empty CIDR to the operator if it was auto-detected and is really unnecessary.